backend/server/settings.py

```diff
@@ -23,6 +20,45 @@ def get_postgres_url() -> URL:
     return url
 
 
-def get_model() -> ChatOpenAI:
+class ChatModel(NamedTuple):
```

Comment:
Have you considered using runnable configurable here?

Reply:
One issue I'm getting with this is if I set model to be `ChatOpenAI(model="gpt-3.5-turbo", temperature=0).configurable_alternatives(...)`, then it is now a `RunnableConfigurableAlternatives` which does not support `with_structured_output`. lmk if you had something else in mind for runnable configurable or if there's a workaround.

Reply:
Added an issue to capture this. Resolving this comment for now.

https://github.com/langchain-ai/langchain/issues/19279